quinn-proto/src/frame.rs

```diff
@@ -403,14 +421,111 @@ impl Ack {
             buf.write_var(size - 1);
             prev = block.start;
         }
-        if let Some(x) = ecn {
+        if let Some(x) = timestamps {
+            Self::encode_timestamps(x.0, largest, buf, x.1, x.2, x.3)
+        } else if let Some(x) = ecn {
             x.encode(buf)
         }
     }
 
     pub fn iter(&self) -> AckIter<'_> {
         self.into_iter()
     }
+
+    pub fn timestamps_iter(&self, epoch: Instant, exponent: u64) -> Option<AckTimestampIter> {
+        self.timestamps
+            .as_ref()
+            .map(|v| AckTimestampIter::new(self.largest, epoch, exponent, &v[..]))
+    }
+
+    // https://www.ietf.org/archive/id/draft-smith-quic-receive-ts-00.html#ts-ranges
+    fn encode_timestamps(
+        timestamps: &ReceiverTimestamps,
+        mut largest: u64,
+        buf: &mut impl BufMut,
+        mut basis: Instant,
+        exponent: u64,
+        max_timestamps: u64,
```

Comment:
This doesn't seem to be used for anything but a `debug_assert`. If we trust the caller to pass the right limit down, we might as well just trust the caller to truncate the timestamp list.